specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json

```diff
@@ -106,7 +106,8 @@
             "description": "The expand expression to apply to the operation.",
             "type": "string",
             "enum": [
-              "instanceView"
+              "instanceView",
+              "userData"
```

Comment:
Where are we going to document that userData is not supported for dedicated host group?

Reply:
@smotwani

Reply:
We should the explanation of all the enum options supported in the description of $expand  and the resources they are applicable to.

Reply:
Unfortunately, CloudService Swagger and CRP Swagger are following an anti-pattern in that they both use the same enum. This means that if 'InstanceViewTypes' is changed in one service, the same change must be made in all services that also use 'InstanceViewTypes'. 

"x-ms-enum": {
              "name": "InstanceViewTypes",

In order to add a new enum value to InstanceViewTypes (or any other existing enum), the enum value has to be added to all instances of InstanceViewTypes.
Otherwise, the PR check and AutoRestGeneration will fail with this error

“Swagger document contains two or more x-ms-enum extensions with the same name 'InstanceViewTypes' and different values: instanceView,userData vs. instanceView”

The dev who created 'InstanceViewTypes' should instead have used a naming pattern like 'ExpandTypesFor{resourceName}' (e.g. 'ExpandTypesForVirtualMachine' or 'ExpandTypesForVirtualMachineScaleSet'), which would have allowed different features to extend the $expand for different scenarios. Unfortunately, we are not stuck with this 'InstanceViewTypes'. 

Any other change (such as changing ModelAsString to true, or adding an entirely new eum) is a breaking change that is very unlikely to be approved by the BreakingChange review board.

Reply:
Okay, update the description in that case to say that userData is not supported.